### DISCUSSION TITLE
shadow-inner sizes

### DISCUSSION THREAD

**Discussion by csvidit** [tailwindlabs/tailwindcss#12537] [Ideas]
It looks like the ```shadow-inner``` class is an alternative for ```shadow-{size}``` classes, which reduces control if someone wants to use shadow-inner in conjunction with shadow-{size} classes (for example, ```<div className="shadow-inner shadow-xl shadow-neutral-100">```). We should make the shadow-inner class an additive class for the ```shadow-{sizes}``` classes, so that the regular ```shadow``` class represents a standard-size outer/drop shadow, ```shadow-inner``` represents a standard inner/inset shadow, and then the ```shadow-{sizes}``` classes can be used to modify the size of either.

**Comment by adamwathan** [comment]
Totally agree and it's on my todo list for a future version — would be nice to be able to set the color of them separately too I think 👌🏻

**Reply by aminmhz** [reply]
1000 years later

**Comment by neochief** [comment]
This would be very useful.

**Comment by neochief** [comment]
I think this is nailed in 4.0: https://tailwindcss.com/docs/box-shadow#adding-an-inset-shadow
